src/ORM/FieldType/DBForeignKey.php

```diff
@@ -63,11 +73,18 @@ public function scaffoldFormField($title = null, $params = null)
         $list = DataList::create($hasOneClass);
         // Don't scaffold a dropdown for large tables, as making the list concrete
         // might exceed the available PHP memory in creating too many DataObject instances
-        if ($list->count() < 100) {
+        $threshold = self::config()->get('dropdown_field_threshold');
+        if ($list->count() < $threshold) {
             $field = new DropdownField($this->name, $title, $list->map('ID', $titleField));
             $field->setEmptyString(' ');
         } else {
             $field = new NumericField($this->name, $title);
+
+            $field->setRightTitle(
+                'This field has been converted to a NumericField as there are at least ' . $threshold
+                . ' related object(s). This can be set with the config variable'
+                . ' SilverStripe\ORM\FieldTypeDBForeignKey::dropdown_field_threshold'
```

Comment:
I don't think that this description belongs on the field, since it's for CMS users rather than developers. This message likely to confuse the user more than anything.  What do you think?

Reply:
Oh, I see it was raised by @kinglozzer too. My vote is to remove this and leave the configurable limit in place without this message.

Reply:
It was intended for phpdoc not for the user, right.

Reply:
Well, no, not really. The problem raised was that the issue was invisible:

>This becomes extremely frustrating from a UX Point of View where the CMS works and looks nice, but if the client adds more than 100 objects then it breaks the experience.

I agree that we shouldn't be exposing the config stuff straight to the user, but it was complicated and not necessarily intuitive to find this particular block of code to figure out what was going on, so I'd argue that _some_ sort of message leaves a much better experience for both the CMS user and the dev.

Even something like "Too many objects for dropdown, fallback used" would help enormously in debugging, and give the user something useful to pass to their dev, who in turn has something to search for when they are looking to fix it.

Otherwise you end up with a field that seems to have mysteriously changed for what is an invisible reason, which is frustrating and confusing.

The other options I see as viable:

- move this functionality to `DropdownField` itself - which is where one would assume that the config var is located if you were to guess
- add some specific documentation around the issue and the config var

Ultimately up to you, but I think making this less "magical" is better